### DISCUSSION TITLE
Darker shade color

### DISCUSSION THREAD

**Discussion by branaway** [pixiv/three-vrm#1083] [Help]
Is there a way to configure the material to carry a darker shade when the the part of the body is not exposed in light? The idea is to make the model look more realistic in night scene.

**Comment by 0b5vr** [comment]
>when the the part of the body is not exposed in light?

Needs more context; Is it whether the part is occluded by an object which casts a shadow, or simply the light source is not bright?

**Reply by branaway** [reply]
Thanks for replying. I meant real shadow caused by occlusion . In a dark scene such as on a stage where there is only one light source. I'd like the part lit by the light source to be bright and the part in the shadow to be darker than the normal shadow color of the toon material.  Better yet, we have a gradient map that we can control the gradient, similar the toon material in Three.js:  https://threejs.org/examples/?q=toon#webgl_materials_variations_toon

**Comment by 0b5vr** [comment]
Unfortunately I think that is the weak point of the MToon, it's difficult to darken occluded parts.
Is anyone have a good way to handle this?

**Reply by branaway** [reply]
https://threejs.org/examples/?q=toon#webgl_materials_variations_toon  offers gradient map that controls the shading quite well. Can we do something similar in the three-vrm? Still another thought, is it possible that we replace the arm toon material with the threejs default toon material in my webgl rendering process?

**Reply by 0b5vr** [reply]
The purpose of the MToon is to maintain visual compatibility between multiple platforms.
You can technically convert MToons into other materials but you should try to keep the look the same.
